### PR TITLE
adafruit-libraries: update all subpackages

### PR DESCRIPTION
**Pull request by InuSasha**
- ~~switch to sha in full length~~
- add script for automatic update
- fix build of DHT package

### CODE REVIEW

**Comment by CvH** [comment]
@InuSasha to avoid the the git user/pw you can use the "public api" - I had this hacky line in stash to ask for the last commit date from git (I am pretty sure this could be simplified a lot)
```
# get last commit date
GIT_REV=$( curl -s https://api.github.com/repos/LibreELEC/LibreELEC.tv/commits/master | tac | tac | head -8 | grep date | awk '{ print $2 }' | sed s/\"//g | sed s/,//g | sed 's/.\{10\}$//');
```
`switch to sha in full length` do we want this ?

**Comment by InuSasha** [comment]
- replaced copyright
- better help text

@CvH 
I used the full sha, because i don't know how github shrink the full. and there is no big human interaction. So it is the easiest way, i think.

The github login is only needed, if we hit the rate limit (after 4 runs of the script, for example. or in combination with other scripts). I changed the help text, to make it more obvious.

**Comment by chewitt** [comment]
@InuSasha the short githash is the first 7 characters of the full hash

**Comment by InuSasha** [comment]
@CvH switched back to short sha
@chewitt thanks

**Comment by chewitt** [comment]
@InuSasha If you import from ~/.gitconfig (as suggested by milhouse in Slack) the maintenance script can be simplified.

**Comment by CvH** [comment]
@InuSasha https://github.com/LibreELEC/LibreELEC.tv/blob/master/tools/distro-tool#L44-L45 was milhouse suggestion

**Comment by InuSasha** [comment]
switched to milhouse solution, for the github credentials

**Comment by MilhouseVH** [comment]
It's ~/.git.conf:
```
GIT_USERNAME="username"
GIT_PASSWORD="password_or_token"
```

File: tools/mkpkg/update_adafruit-libraries [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  GIT_HASH=$(curl --silent --header "Accept: application/vnd.github.VERSION.sha" $CURL_LOGIN $URL)
+  
+  # check rate limit
+  if [[ "$GIT_HASH" == *rate\ limit* ]]; then
+    echo "abort, rate limit of api reached. try later or use your github account"
+    exit 1
+  fi
+  
+  # compare hash
+  GIT_HASH=$(cut -c -7 <<< $GIT_HASH)
```
Comment (MilhouseVH) [inline tools/mkpkg/update_adafruit-libraries]: `GIT_HASH=${GIT_HASH:0:7}` is easier.

File: tools/mkpkg/update_adafruit-libraries [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  if [ $PKG_VERSION == $GIT_HASH ]; then
+    echo "unchanged"
+    continue
+  fi
+
+  # replace old with new hash
+  sed "s|^PKG_VERSION=.*$|PKG_VERSION=\"${GIT_HASH}\"|" \
```
Comment (MilhouseVH) [inline tools/mkpkg/update_adafruit-libraries]: `sed "s|^PKG_VERSION=.*$|PKG_VERSION=\"${GIT_HASH}\"|" -i ${PKG_DIR}/package.mk`
then you can drop the following 3 lines (72-74).

File: tools/mkpkg/update_adafruit-libraries [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  CHANGED="${CHANGED}#$PKG_SHORTDESC"
+done
+
+# prepare revision and changelog
+. config/options adafruit-libraries
+if [ ! -z "$CHANGED" ]; then
+  # revision
+  REV=$(( $PKG_REV + 1 ))
+  echo "increase revision $PKG_REV -> $REV"
+  sed "s|^PKG_REV=.*$|PKG_REV=\"${REV}\"|" \
```
Comment (MilhouseVH) [inline tools/mkpkg/update_adafruit-libraries]: Same as above.

File: tools/mkpkg/update_adafruit-libraries [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  
+  # filter out, all non github packages
+  if [[ $PKG_URL != https://github.com/* ]]; then
+    echo "ignore, not on github"
+    continue
+  fi
+  
+  # get new hash from master branch
+  URL=$(sed "s|\(www\.\)*github.com/|api.github.com/repos/|;s|/archive/.*|/commits/master|" <<< $PKG_URL)
+  GIT_HASH=$(curl --silent --header "Accept: application/vnd.github.VERSION.sha" $CURL_LOGIN $URL)
```
Comment (MilhouseVH) [inline tools/mkpkg/update_adafruit-libraries]: `VERSION` should be `v3` rather than the literal VERSION: https://developer.github.com/v3/media/

**Comment by InuSasha** [comment]
implement @MilhouseVH suggestions, and feel like a rookie :)

**Comment by InuSasha** [comment]
add SHA256 hashes to packages and update script

**Pull request merged**